qiskit/optimization/algorithms/admm_optimizer.py

```diff
@@ -0,0 +1,897 @@
+
+        # constraints for z.
+        # A1 z <= b1.
+        constraint_count = self._state.a1.shape[0]
+        for i in range(constraint_count):
+            linear = np.concatenate((np.zeros(continuous_size), self._state.a1[i, :]))
+            op2.linear_constraint(linear=linear, sense=Constraint.Sense.LE, rhs=self._state.b1[i])
+
+        if continuous_size:
+            # A2 z + A3 u <= b2
+            constraint_count = self._state.a2.shape[0]
+            for i in range(constraint_count):
+                linear = np.concatenate((self._state.a3[i, :], self._state.a2[i, :]))
+                op2.linear_constraint(linear=linear,
+                                      sense=Constraint.Sense.LE,
+                                      rhs=self._state.b2[i])
+
+            # A4 u <= b3
+            constraint_count = self._state.a4.shape[0]
+            for i in range(constraint_count):
+                linear = np.concatenate((self._state.a4[i, :], np.zeros(binary_size)))
+                op2.linear_constraint(linear=linear,
+                                      sense=Constraint.Sense.LE,
+                                      rhs=self._state.b3[i])
+
+        # add quadratic constraints for
+        # In the step 2, we basically need to copy all quadratic constraints of the original
+        # problem, and substitute binary variables with variables 0<=z<=1.
+        # The quadratic constraint can have any equality/inequality sign.
+        #
+        #  For example:
+        #  Original problem:
+        #  Quadratic_constraint: x**2 + u**2 <= 1
+        #  Vars: x binary, L <= u <= U
+        #
+        #  Step 2:
+        #  Quadratic_constraint: z**2 + u**2 <= 1
+        #  Vars: 0<=z<=1, L <= u <= U
+        # for linear, quad, sense, rhs \
+        #         in zip(self._state.op.quadratic_constraints.get_linear_components(),
+        #                self._state.op.quadratic_constraints.get_quadratic_components(),
+        #                self._state.op.quadratic_constraints.get_senses(),
+        #                self._state.op.quadratic_constraints.get_rhs()):
+        #     # types in the loop: SparsePair, SparseTriple, character, float
+        #     print(linear, quad, sense, rhs)
+        #     new_linear = SparsePair(ind=self._binary_indices_to_continuous(linear.ins),
+        #                             val=linear.val)
+        #     new_quadratic = SparseTriple(ind1=self._binary_indices_to_continuous(quad.ind1),
+        #                                  ind2=self._binary_indices_to_continuous(quad.ind2),
+        #                                  val=quad.val)
+        #     op2.quadratic_constraints.add(lin_expr=new_linear, quad_expr=new_quadratic,
+        #                                   sense=sense, rhs=rhs)
+
+        return op2
+
+    def _binary_indices_to_continuous(self, binary_indices: List[int]) -> List[int]:
+        # TODO: implement
```

Comment:
If this is unimplemented still is it better to raise an error here in the meanwhile

Reply:
Definitely, agreed